documentation/contracts.md

```diff
@@ -42,16 +44,188 @@ While a more detailed response could look like this:
 }
 ```
 
-The endpoint should return a `200 OK` if the service is possibly healthy (`UP`, `UNKNOWN`), and a `503 Service Unavailable` if the service is unhealthy (`DOWN`, `OUT_OF_SERVICE`).
+The endpoint MUST return a `200 OK` if the service is possibly healthy (`UP`, `UNKNOWN`), and a `503 Service Unavailable` if the service is unhealthy (`DOWN`, `OUT_OF_SERVICE`).
 
-## Endpoints route
+## Error schema
 
-TODO
+The JSON objects returned alongside `4xx` and `5xx` HTTP status codes SHOULD follow the following schema:
 
-## Error schema
+```JSONC
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "title": "AppError",
+  "description": "API error responses for the JakPrzyjade system.",
+  "type": "object",
+  "readOnly": true,
+  "properties": {
+    "code": { // due to layer separation, we should pretend that this is not HTTP-related, but truly, it is
+      "type": "integer",
+      "description": "HTTP error status code.",
+      "minimum": 400,
+      "maximum": 599
+    },
+    "kind": { // identifier for the error, used for logging and client-side error handling
+      "type": "string",
+      "description": "Error kind, lowercase kebab-case, without trailing '-error'.",
+      "maxLength": 255,
+      "pattern": "^[a-z]+(-[a-z]+)*$" // kebab-case
+    },
+    "messageEn": { // human-readable error message in English
+      "type": "string",
+      "description": "Error message in English."
+    },
+    "messagePl": { // human-readable error message in Polish, only required for user-facing errors
+      "type": "string",
+      "description": "Error message in Polish."
+    }
+  },
+  "required": ["code", "kind", "messageEn"] // "messagePl" is optional
+}
+```
+
+For example:
+
+```JSONC
+{
+  "code": 404,
+  "kind": "account-not-found",
+  "messageEn": "Account with ID {0193f896-1452-71d3-8721-64810efa81ae} was not found!",
+  "messagePl": "Konto o ID {0193f896-1452-71d3-8721-64810efa81ae} nie zostało znalezione!"
+}
+```
+
+Which translates to the following types in the languages used in the project:
+
+```TypeScript
+type AppError = {
+  code: number; // const literal
+  kind: string; // const literal
+  messageEn: string;
+  messagePl?: string;
+};
+```
+
+```Java
+@Data
+@AllArgsConstructor
+public class AppError {
+    @NotNull private final int code;
+    @NotNull private final String kind;
+    @NotNull private final String messageEn;
+    private final String messagePl;
+}
+```
```

Comment:
Nit: in the documentation, I would not use the lombok annotation, and maybe even use record here instead of class

Reply:
Also `int` can't be null so `@NotNull` is not necessary here, if not even invalid